src/sql/magic.py

```diff
@@ -410,18 +414,43 @@ def interactive_execute_wrapper(**kwargs):
             else:
                 with_ = self._store.infer_dependencies(command.sql_original, args.save)
                 if with_:
-                    command.set_sql_with(with_)
-                    display.message(
-                        f"Generating CTE with stored snippets: {pretty_print(with_)}"
-                    )
+                    query_type = self.get_query_type(command, original=True)
+
+                    if query_type != "SELECT":
+                        display.message_warning(
+                            f"Cannot use snippets with {query_type}"
```

Comment:
the message should be clearer, something like:

```
Your query is using the identifiers 'a', 'b', 'c', which are stored snippets, but snippets only work with SELECT statements, hence, CTE generation is disabled
```

Reply:
I've added better warning messages